Travis CI native debugger compilation error

We didn't caught them earlier because Travis CI service wasn't running yesterday.

Linux on System Z (S390):
```
usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_gcore':
/home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:793: undefined reference to `linux_generate_corefile'
/usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_native_bp':
/home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:740: undefined reference to `arm64_hwbp_add'
/usr/bin/ld: /home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:741: undefined reference to `arm64_hwbp_del'
collect2: error: ld returned 1 exit status
````
https://app.travis-ci.com/github/rizinorg/rizin/jobs/625811858

Linux on PPC:
```
[1878/2275] Compiling C object librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o
In file included from ../librz/debug/p/native/linux_arm64.c:14,
                 from ../librz/debug/p/debug_native.c:46:
../librz/debug/p/native/linux/linux_coredump.h:63: warning: "RZ_DEBUG_REG_T" redefined
   63 | #define RZ_DEBUG_REG_T struct user_regs_struct
      | 
In file included from ../librz/debug/p/native/linux_arm64.c:12,
                from ../librz/debug/p/debug_native.c:46:
../librz/debug/p/native/linux/linux_debug.h:132: note: this is the location of the previous definition
  132 | #define RZ_DEBUG_REG_T struct powerpc_regs_t
      | 
In file included from ../librz/debug/p/debug_native.c:46:
../librz/debug/p/native/linux_arm64.c: In function ‘rz_debug_native_bp’:
../librz/debug/p/native/linux_arm64.c:740:6: warning: implicit declaration of function ‘arm64_hwbp_add’ [-Wimplicit-function-declaration]
  740 |    ? arm64_hwbp_add((RzDebug *)bp->user, bp, b)
      |      ^~~~~~~~~~~~~~
../librz/debug/p/native/linux_arm64.c:741:6: warning: implicit declaration of function ‘arm64_hwbp_del’ [-Wimplicit-function-declaration]
  741 |    : arm64_hwbp_del((RzDebug *)bp->user, bp, b);
      |      ^~~~~~~~~~~~~~
At top level:
../librz/debug/p/native/linux_arm64.c:658:13: warning: ‘ll_arm32_hwbp_set’ defined but not used [-Wunused-function]
  658 | static bool ll_arm32_hwbp_set(pid_t pid, ut64 addr, int size, int wp, int type) {
      |             ^~~~~~~~~~~~~~~~~
[1879/2275] Compiling C object librz/debug/librz_debug.so.0.8.0.p/p_native_procfs.c.o
[1880/2275] Compiling C object librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_debug.c.o
../librz/debug/p/native/linux/linux_debug.c: In function ‘print_fpu’:
../librz/debug/p/native/linux/linux_debug.c:1146:2: warning: #warning print_fpu not implemented for this platform [-Wcpp]
 1146 | #warning print_fpu not implemented for this platform
      |  ^~~~~~~
../librz/debug/p/native/linux/linux_debug.c: In function ‘linux_reg_read’:
../librz/debug/p/native/linux/linux_debug.c:1151:7: warning: variable ‘showfpu’ set but not used [-Wunused-but-set-variable]
 1151 |  bool showfpu = false;
      |       ^~~~~~~
../librz/debug/p/native/linux/linux_debug.c: In function ‘linux_reg_write’:
../librz/debug/p/native/linux/linux_debug.c:1354:55: warning: passing argument 4 of ‘rz_debug_ptrace’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1354 |   int ret = rz_debug_ptrace(dbg, PTRACE_SETREGS, pid, buf, NULL);
      |                                                       ^~~
In file included from ../librz/debug/p/native/linux/linux_debug.c:6:
../librz/include/rz_debug.h:611:96: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  611 | static inline long rz_debug_ptrace(RzDebug *dbg, rz_ptrace_request_t request, pid_t pid, void *addr, rz_ptrace_data_t data) {
      |                                                                                          ~~~~~~^~~~
At top level:
../librz/debug/p/native/linux/linux_debug.c:1057:13: warning: ‘print_fpu’ defined but not used [-Wunused-function]
 1057 | static void print_fpu(void *f) {
      |             ^~~~~~~~~
[1881/2275] Compiling C object subprojects/rizin-shell-parser/libshell_parser.a.p/src_scanner.c.o
[1882/2275] Linking target librz/debug/librz_debug.so.0.8.0
FAILED: librz/debug/librz_debug.so.0.8.0 

gcc  -o librz/debug/librz_debug.so.0.8.0 librz/debug/librz_debug.so.0.8.0.p/ddesc.c.o librz/debug/librz_debug.so.0.8.0.p/debug.c.o librz/debug/librz_debug.so.0.8.0.p/dreg.c.o librz/debug/librz_debug.so.0.8.0.p/desil.c.o librz/debug/librz_debug.so.0.8.0.p/dmap.c.o librz/debug/librz_debug.so.0.8.0.p/pid.c.o librz/debug/librz_debug.so.0.8.0.p/plugin.c.o librz/debug/librz_debug.so.0.8.0.p/dsession.c.o librz/debug/librz_debug.so.0.8.0.p/dsignal.c.o librz/debug/librz_debug.so.0.8.0.p/serialize_debug.c.o librz/debug/librz_debug.so.0.8.0.p/snap.c.o librz/debug/librz_debug.so.0.8.0.p/trace.c.o librz/debug/librz_debug.so.0.8.0.p/p_bfvm.c.o librz/debug/librz_debug.so.0.8.0.p/p_common_windows.c.o librz/debug/librz_debug.so.0.8.0.p/p_common_winkd.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_bf.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_bochs.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_dmp.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_gdb.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_io.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_null.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_rap.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_winkd.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_qnx.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_debug.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_coredump.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_procfs.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,librz_debug.so.0.8 -DRZ_ASSERT_STDOUT=1 '-Wl,-rpath,$ORIGIN/../util:$ORIGIN/../cons:$ORIGIN/../socket:$ORIGIN/../hash:$ORIGIN/../crypto:$ORIGIN/../io:$ORIGIN/../bin:$ORIGIN/../demangler:$ORIGIN/../magic:$ORIGIN/../syscall:$ORIGIN/../type:$ORIGIN/../reg:$ORIGIN/../bp:$ORIGIN/../egg:$ORIGIN/../arch:$ORIGIN/../search:$ORIGIN/../config:$ORIGIN/../flag:$ORIGIN/../diff:$ORIGIN/../il' -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/util -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/cons -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/socket -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/hash -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/crypto -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/io -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/bin -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/demangler -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/magic -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/syscall -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/type -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/reg -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/bp -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/egg -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/arch -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/search -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/config -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/flag -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/diff -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/il -Wl,--start-group subprojects/rzgdb/librzgdb.a librz/util/librz_util.so.0.8.0 librz/cons/librz_cons.so.0.8.0 librz/socket/librz_socket.so.0.8.0 subprojects/rzwinkd/librzwinkd.a librz/hash/librz_hash.so.0.8.0 librz/crypto/librz_crypto.so.0.8.0 librz/io/librz_io.so.0.8.0 librz/bin/librz_bin.so.0.8.0 librz/reg/librz_reg.so.0.8.0 librz/bp/librz_bp.so.0.8.0 librz/syscall/librz_syscall.so.0.8.0 librz/egg/librz_egg.so.0.8.0 librz/arch/librz_arch.so.0.8.0 librz/type/librz_type.so.0.8.0 subprojects/rzqnx/librzqnx.a -Wl,--end-group
/usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_gcore':
/home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:793: undefined reference to `linux_generate_corefile'
/usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_native_bp':
/home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:741: undefined reference to `arm64_hwbp_del'
/usr/bin/ld: /home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:740: undefined reference to `arm64_hwbp_add'
collect2: error: ld returned 1 exit status
```
https://app.travis-ci.com/github/rizinorg/rizin/jobs/625811857


Linux on ARM64:
```
../librz/debug/p/native/linux/linux_debug.c: In function ‘print_fpu’:
../librz/debug/p/native/linux/linux_debug.c:1146:2: warning: #warning print_fpu not implemented for this platform [-Wcpp]
 1146 | #warning print_fpu not implemented for this platform
      |  ^~~~~~~
../librz/debug/p/native/linux/linux_debug.c: In function ‘linux_reg_read’:
../librz/debug/p/native/linux/linux_debug.c:1250:2: warning: #warning getfpregs not implemented for this platform [-Wcpp]
 1250 | #warning getfpregs not implemented for this platform
      |  ^~~~~~~
../librz/debug/p/native/linux/linux_debug.c:1151:7: warning: variable ‘showfpu’ set but not used [-Wunused-but-set-variable]
 1151 |  bool showfpu = false;
      |       ^~~~~~~
At top level:
../librz/debug/p/native/linux/linux_debug.c:1057:13: warning: ‘print_fpu’ defined but not used [-Wunused-function]
 1057 | static void print_fpu(void *f) {
      |             ^~~~~~~~~
[1880/2275] Compiling C object librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o
In file included from ../librz/debug/p/native/linux_arm64.c:14,
                 from ../librz/debug/p/debug_native.c:46:
../librz/debug/p/native/linux/linux_coredump.h:63: warning: "RZ_DEBUG_REG_T" redefined
   63 | #define RZ_DEBUG_REG_T struct user_regs_struct
      | 
In file included from ../librz/debug/p/native/linux_arm64.c:12,
                 from ../librz/debug/p/debug_native.c:46:
../librz/debug/p/native/linux/linux_debug.h:106: note: this is the location of the previous definition
  106 | #define RZ_DEBUG_REG_T struct user_pt_regs
      | 
In file included from ../librz/debug/p/debug_native.c:46:
../librz/debug/p/native/linux_arm64.c: In function ‘rz_debug_native_bp’:
../librz/debug/p/native/linux_arm64.c:740:6: warning: implicit declaration of function ‘arm64_hwbp_add’ [-Wimplicit-function-declaration]
  740 |    ? arm64_hwbp_add((RzDebug *)bp->user, bp, b)
      |      ^~~~~~~~~~~~~~
../librz/debug/p/native/linux_arm64.c:741:6: warning: implicit declaration of function ‘arm64_hwbp_del’ [-Wimplicit-function-declaration]
  741 |    : arm64_hwbp_del((RzDebug *)bp->user, bp, b);
      |      ^~~~~~~~~~~~~~
At top level:
../librz/debug/p/native/linux_arm64.c:658:13: warning: ‘ll_arm32_hwbp_set’ defined but not used [-Wunused-function]
  658 | static bool ll_arm32_hwbp_set(pid_t pid, ut64 addr, int size, int wp, int type) {
      |             ^~~~~~~~~~~~~~~~~
[1881/2275] Linking target librz/debug/librz_debug.so.0.8.0
FAILED: librz/debug/librz_debug.so.0.8.0 

gcc  -o librz/debug/librz_debug.so.0.8.0 librz/debug/librz_debug.so.0.8.0.p/ddesc.c.o librz/debug/librz_debug.so.0.8.0.p/debug.c.o librz/debug/librz_debug.so.0.8.0.p/dreg.c.o librz/debug/librz_debug.so.0.8.0.p/desil.c.o librz/debug/librz_debug.so.0.8.0.p/dmap.c.o librz/debug/librz_debug.so.0.8.0.p/pid.c.o librz/debug/librz_debug.so.0.8.0.p/plugin.c.o librz/debug/librz_debug.so.0.8.0.p/dsession.c.o librz/debug/librz_debug.so.0.8.0.p/dsignal.c.o librz/debug/librz_debug.so.0.8.0.p/serialize_debug.c.o librz/debug/librz_debug.so.0.8.0.p/snap.c.o librz/debug/librz_debug.so.0.8.0.p/trace.c.o librz/debug/librz_debug.so.0.8.0.p/p_bfvm.c.o librz/debug/librz_debug.so.0.8.0.p/p_common_windows.c.o librz/debug/librz_debug.so.0.8.0.p/p_common_winkd.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_bf.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_bochs.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_dmp.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_gdb.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_io.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_null.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_rap.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_winkd.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_qnx.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_debug.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_coredump.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_procfs.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,librz_debug.so.0.8 -DRZ_ASSERT_STDOUT=1 '-Wl,-rpath,$ORIGIN/../util:$ORIGIN/../cons:$ORIGIN/../socket:$ORIGIN/../hash:$ORIGIN/../crypto:$ORIGIN/../io:$ORIGIN/../bin:$ORIGIN/../demangler:$ORIGIN/../magic:$ORIGIN/../syscall:$ORIGIN/../type:$ORIGIN/../reg:$ORIGIN/../bp:$ORIGIN/../egg:$ORIGIN/../arch:$ORIGIN/../search:$ORIGIN/../config:$ORIGIN/../flag:$ORIGIN/../diff:$ORIGIN/../il' -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/util -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/cons -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/socket -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/hash -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/crypto -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/io -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/bin -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/demangler -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/magic -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/syscall -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/type -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/reg -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/bp -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/egg -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/arch -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/search -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/config -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/flag -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/diff -Wl,-rpath-link,/home/travis/build/rizinorg/rizin/build/librz/il -Wl,--start-group subprojects/rzgdb/librzgdb.a librz/util/librz_util.so.0.8.0 librz/cons/librz_cons.so.0.8.0 librz/socket/librz_socket.so.0.8.0 subprojects/rzwinkd/librzwinkd.a librz/hash/librz_hash.so.0.8.0 librz/crypto/librz_crypto.so.0.8.0 librz/io/librz_io.so.0.8.0 librz/bin/librz_bin.so.0.8.0 librz/reg/librz_reg.so.0.8.0 librz/bp/librz_bp.so.0.8.0 librz/syscall/librz_syscall.so.0.8.0 librz/egg/librz_egg.so.0.8.0 librz/arch/librz_arch.so.0.8.0 librz/type/librz_type.so.0.8.0 subprojects/rzqnx/librzqnx.a -Wl,--end-group

/usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_gcore':
/home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:793: undefined reference to `linux_generate_corefile'
/usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_native_bp':
/home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:740: undefined reference to `arm64_hwbp_add'
/usr/bin/ld: /home/travis/build/rizinorg/rizin/build/../librz/debug/p/native/linux_arm64.c:741: undefined reference to `arm64_hwbp_del'

collect2: error: ld returned 1 exit status
```
https://app.travis-ci.com/github/rizinorg/rizin/jobs/625811859

cc @tushar3q34 @wargio 

It also breaks Docker builds:
```
[1883/2273] Compiling C object librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_debug.c.o
#40 2458.1 ../***/librz/debug/p/native/linux/linux_debug.c: In function ‘print_fpu’:
#40 2458.1 ../***/librz/debug/p/native/linux/linux_debug.c:1146:2: warning: #warning print_fpu not implemented for this platform [-Wcpp]
#40 2458.1  #warning print_fpu not implemented for this platform
#40 2458.1   ^~~~~~~
#40 2458.1 ../***/librz/debug/p/native/linux/linux_debug.c: In function ‘linux_reg_read’:
#40 2458.1 ../***/librz/debug/p/native/linux/linux_debug.c:1250:2: warning: #warning getfpregs not implemented for this platform [-Wcpp]
#40 2458.1  #warning getfpregs not implemented for this platform
#40 2458.1   ^~~~~~~
#40 2458.1 ../***/librz/debug/p/native/linux/linux_debug.c:1151:7: warning: variable ‘showfpu’ set but not used [-Wunused-but-set-variable]
#40 2458.1   bool showfpu = false;
#40 2458.1        ^~~~~~~
#40 2458.1 At top level:
#40 2458.1 ../***/librz/debug/p/native/linux/linux_debug.c:1057:13: warning: ‘print_fpu’ defined but not used [-Wunused-function]
#40 2458.1  static void print_fpu(void *f) {
#40 2458.1              ^~~~~~~~~
#40 2459.2 [1884/2273] Compiling C object librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o
#40 2459.2 In file included from ../***/librz/debug/p/native/linux_arm64.c:14,
#40 2459.2                  from ../***/librz/debug/p/debug_native.c:46:
#40 2459.2 ../***/librz/debug/p/native/linux/linux_coredump.h:63: warning: "RZ_DEBUG_REG_T" redefined
#40 2459.2  #define RZ_DEBUG_REG_T struct user_regs_struct
#40 2459.2  
#40 2459.2 In file included from ../***/librz/debug/p/native/linux_arm64.c:12,
#40 2459.2                  from ../***/librz/debug/p/debug_native.c:46:
#40 2459.2 ../***/librz/debug/p/native/linux/linux_debug.h:106: note: this is the location of the previous definition
#40 2459.2  #define RZ_DEBUG_REG_T struct user_pt_regs
#40 2459.2  
#40 2459.2 In file included from ../***/librz/debug/p/debug_native.c:46:
#40 2459.2 ../***/librz/debug/p/native/linux_arm64.c: In function ‘rz_debug_native_bp’:
#40 2459.2 ../***/librz/debug/p/native/linux_arm64.c:740:6: warning: implicit declaration of function ‘arm64_hwbp_add’; did you mean ‘rz_bv_add’? [-Wimplicit-function-declaration]
#40 2459.2     ? arm64_hwbp_add((RzDebug *)bp->user, bp, b)
#40 2459.2       ^~~~~~~~~~~~~~
#40 2459.2       rz_bv_add
#40 2459.2 ../***/librz/debug/p/native/linux_arm64.c:741:6: warning: implicit declaration of function ‘arm64_hwbp_del’; did you mean ‘rz_bp_del’? [-Wimplicit-function-declaration]
#40 2459.2     : arm64_hwbp_del((RzDebug *)bp->user, bp, b);
#40 2459.2       ^~~~~~~~~~~~~~
#40 2459.2       rz_bp_del
#40 2459.2 At top level:
#40 2459.2 ../***/librz/debug/p/native/linux_arm64.c:658:13: warning: ‘ll_arm32_hwbp_set’ defined but not used [-Wunused-function]
#40 2459.2  static bool ll_arm32_hwbp_set(pid_t pid, ut64 addr, int size, int wp, int type) {
#40 2459.2              ^~~~~~~~~~~~~~~~~
#40 2460.6 [1885/2273] Linking target librz/debug/librz_debug.so.0.8.0
#40 2460.6 FAILED: librz/debug/librz_debug.so.0.8.0 
#40 2460.6 cc  -o librz/debug/librz_debug.so.0.8.0 librz/debug/librz_debug.so.0.8.0.p/ddesc.c.o librz/debug/librz_debug.so.0.8.0.p/debug.c.o librz/debug/librz_debug.so.0.8.0.p/dreg.c.o librz/debug/librz_debug.so.0.8.0.p/desil.c.o librz/debug/librz_debug.so.0.8.0.p/dmap.c.o librz/debug/librz_debug.so.0.8.0.p/pid.c.o librz/debug/librz_debug.so.0.8.0.p/plugin.c.o librz/debug/librz_debug.so.0.8.0.p/dsession.c.o librz/debug/librz_debug.so.0.8.0.p/dsignal.c.o librz/debug/librz_debug.so.0.8.0.p/serialize_debug.c.o librz/debug/librz_debug.so.0.8.0.p/snap.c.o librz/debug/librz_debug.so.0.8.0.p/trace.c.o librz/debug/librz_debug.so.0.8.0.p/p_bfvm.c.o librz/debug/librz_debug.so.0.8.0.p/p_common_windows.c.o librz/debug/librz_debug.so.0.8.0.p/p_common_winkd.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_bf.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_bochs.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_dmp.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_gdb.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_io.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_null.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_rap.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_winkd.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_qnx.c.o librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_debug.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_linux_linux_coredump.c.o librz/debug/librz_debug.so.0.8.0.p/p_native_procfs.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,librz_debug.so.0.8 '-Wl,-rpath,$ORIGIN/../util:$ORIGIN/../cons:$ORIGIN/../socket:$ORIGIN/../hash:$ORIGIN/../crypto:$ORIGIN/../io:$ORIGIN/../bin:$ORIGIN/../demangler:$ORIGIN/../magic:$ORIGIN/../syscall:$ORIGIN/../type:$ORIGIN/../reg:$ORIGIN/../bp:$ORIGIN/../egg:$ORIGIN/../arch:$ORIGIN/../search:$ORIGIN/../config:$ORIGIN/../flag:$ORIGIN/../diff:$ORIGIN/../il' -Wl,-rpath-link,/tmp/build/librz/util -Wl,-rpath-link,/tmp/build/librz/cons -Wl,-rpath-link,/tmp/build/librz/socket -Wl,-rpath-link,/tmp/build/librz/hash -Wl,-rpath-link,/tmp/build/librz/crypto -Wl,-rpath-link,/tmp/build/librz/io -Wl,-rpath-link,/tmp/build/librz/bin -Wl,-rpath-link,/tmp/build/librz/demangler -Wl,-rpath-link,/tmp/build/librz/magic -Wl,-rpath-link,/tmp/build/librz/syscall -Wl,-rpath-link,/tmp/build/librz/type -Wl,-rpath-link,/tmp/build/librz/reg -Wl,-rpath-link,/tmp/build/librz/bp -Wl,-rpath-link,/tmp/build/librz/egg -Wl,-rpath-link,/tmp/build/librz/arch -Wl,-rpath-link,/tmp/build/librz/search -Wl,-rpath-link,/tmp/build/librz/config -Wl,-rpath-link,/tmp/build/librz/flag -Wl,-rpath-link,/tmp/build/librz/diff -Wl,-rpath-link,/tmp/build/librz/il -Wl,--start-group subprojects/rzgdb/librzgdb.a librz/util/librz_util.so.0.8.0 librz/cons/librz_cons.so.0.8.0 librz/socket/librz_socket.so.0.8.0 subprojects/rzwinkd/librzwinkd.a librz/hash/librz_hash.so.0.8.0 librz/crypto/librz_crypto.so.0.8.0 librz/io/librz_io.so.0.8.0 librz/bin/librz_bin.so.0.8.0 librz/reg/librz_reg.so.0.8.0 librz/bp/librz_bp.so.0.8.0 librz/syscall/librz_syscall.so.0.8.0 librz/egg/librz_egg.so.0.8.0 librz/arch/librz_arch.so.0.8.0 librz/type/librz_type.so.0.8.0 subprojects/rzqnx/librzqnx.a -Wl,--end-group
#40 2460.6 /usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_gcore':
#40 2460.6 /tmp/build/../***/librz/debug/p/native/linux_arm64.c:793: undefined reference to `linux_generate_corefile'
#40 2460.6 /usr/bin/ld: librz/debug/librz_debug.so.0.8.0.p/p_debug_native.c.o: in function `rz_debug_native_bp':
#40 2460.6 /tmp/build/../***/librz/debug/p/native/linux_arm64.c:740: undefined reference to `arm64_hwbp_add'
#40 2460.6 /usr/bin/ld: /tmp/build/../***/librz/debug/p/native/linux_arm64.c:741: undefined reference to `arm64_hwbp_del'
#40 2460.6 collect2: error: ld returned 1 exit status
#40 2462.3 [1886/2273] Compiling C object subprojects/***-shell-parser/libshell_parser.a.p/src_scanner.c.o
#40 2462.5 [1887/2273] Compiling C object librz/core/librz_core.so.0.8.0.p/analysis_objc.c.o
#40 2464.4 [1888/2273] Compiling C object librz/core/librz_core.so.0.8.0.p/analysis_tp.c.o
#40 2467.0 [1889/2273] Compiling C object subprojects/***-shell-parser/libshell_parser.a.p/src_parser.c.o
#40 2474.0 [1890/2273] Compiling C object librz/core/librz_core.so.0.8.0.p/agraph.c.o
#40 2474.0 ninja: build stopped: subcommand failed.
#40 2474.0 INFO: autodetecting backend as ninja
#40 2474.0 INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/build
#40 ERROR: process "/bin/sh -c meson setup --prefix=/usr -Dinstall_sigdb=true /tmp/build && \tmeson compile -C /tmp/build && \tmeson install --destdir /tmp/***-install -C /tmp/build" did not complete successfully: exit code: 1
------
 > [linux/arm64 stage-0  8/15] RUN meson setup --prefix=/usr -Dinstall_sigdb=true /tmp/build && 	meson compile -C /tmp/build && 	meson install --destdir /tmp/***-install -C /tmp/build:
2460.6 /usr/bin/ld: /tmp/build/../***/librz/debug/p/native/linux_arm64.c:741: undefined reference to `arm64_hwbp_del'
2460.6 collect2: error: ld returned 1 exit status
```
See https://github.com/rizinorg/rizin/actions/runs/10706760347/job/29700429546#step:9:17440

--- a/librz/debug/p/debug_native.c
+++ b/librz/debug/p/debug_native.c
@@ -50,12 +50,12 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #include "native/linux_x86_64.c"
 #elif __arm__
 #include "native/linux_arm.c"
-#elif __arm64__ || __aarch64__
+#elif __arm64__
 #include "native/linux_arm64.c"
+#elif __aarch64__
+#include "native/linux_aarch64.c"
 #else
-#warning "Unsupported debugging platform"
-#undef DEBUGGER
-#define DEBUGGER 0
+#include "native/linux_other.c"
 #endif
 
 #elif __WINDOWS__

--- a/librz/debug/p/debug_native.c
+++ b/librz/debug/p/debug_native.c
@@ -24,8 +24,12 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #if __APPLE__
 #if __i386__ || __x86_64__
 #include "native/apple_x86_64.c"
-#else
+#elif __arm64__ || __aarch64__
 #include "native/apple_aarch64.c"
+#else
+#warning "Unsupported debugging platform"
+#undef DEBUGGER
+#define DEBUGGER 0
 #endif
 
 #elif __ANDROID__
@@ -33,8 +37,12 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #include "native/android_x86_64.c"
 #elif __arm__
 #include "native/android_arm.c"
-#else
+#elif __arm64__ || __aarch64__
 #include "native/android_arm64.c"
+#else
+#warning "Unsupported debugging platform"
+#undef DEBUGGER
+#define DEBUGGER 0
 #endif
 
 #elif __linux__
@@ -42,8 +50,12 @@ static int rz_debug_native_reg_write(RzDebug *dbg, int type, const ut8 *buf, int
 #include "native/linux_x86_64.c"
 #elif __arm__
 #include "native/linux_arm.c"
-#else
+#elif __arm64__ || __aarch64__
 #include "native/linux_arm64.c"
+#else
+#warning "Unsupported debugging platform"
+#undef DEBUGGER
+#define DEBUGGER 0
 #endif
 
 #elif __WINDOWS__

--- a/librz/debug/p/native/linux_aarch64.c
+++ b/librz/debug/p/native/linux_aarch64.c
@@ -11,7 +11,6 @@
 #include <sys/mman.h>
 #include "linux/linux_debug.h"
 #include "procfs.h"
-#include "linux/linux_coredump.h"
 
 #ifdef __WALL
 #define WAITPID_FLAGS __WALL
@@ -782,6 +781,5 @@ static int rz_debug_desc_native_open(const char *path) {
 }
 
 static bool rz_debug_gcore(RzDebug *dbg, char *path, RzBuffer *dest) {
-	(void)path;
-	return linux_generate_corefile(dbg, dest);
+	return false;
 }

--- a/librz/debug/p/native/linux_arm64.c
+++ b/librz/debug/p/native/linux_arm64.c
@@ -655,7 +655,6 @@ static int rz_debug_native_drx(RzDebug *dbg, int n, ut64 addr, int sz, int rwx, 
 #define PTRACE_SETHBPREGS 30
 #endif
 
-
 #if PTRACE_GETREGSET
 // type = 2 = write
 // static volatile uint8_t var[96] __attribute__((__aligned__(32)));

--- a/librz/debug/p/native/linux_arm64.c
+++ b/librz/debug/p/native/linux_arm64.c
@@ -655,14 +655,6 @@ static int rz_debug_native_drx(RzDebug *dbg, int n, ut64 addr, int sz, int rwx, 
 #define PTRACE_SETHBPREGS 30
 #endif
 
-static bool ll_arm32_hwbp_set(pid_t pid, ut64 addr, int size, int wp, int type) {
-	const unsigned byte_mask = (1 << size) - 1;
-	// const unsigned type = 2; // Write.
-	const unsigned enable = 1;
-	const unsigned control = byte_mask << 5 | type << 3 | enable;
-	(void)ptrace(PTRACE_SETHBPREGS, pid, -1, (void *)(size_t)addr);
-	return ptrace(PTRACE_SETHBPREGS, pid, -2, &control) != -1;
-}
 
 #if PTRACE_GETREGSET
 // type = 2 = write

--- a/librz/debug/p/native/linux_other.c
+++ b/librz/debug/p/native/linux_other.c
@@ -11,7 +11,6 @@
 #include <sys/mman.h>
 #include "linux/linux_debug.h"
 #include "procfs.h"
-#include "linux/linux_coredump.h"
 
 #ifdef __WALL
 #define WAITPID_FLAGS __WALL
@@ -641,98 +640,7 @@ static int rz_debug_native_drx(RzDebug *dbg, int n, ut64 addr, int sz, int rwx, 
 	return -1;
 }
 
-#include <sys/prctl.h>
-#include <sys/uio.h>
-
-#define NT_ARM_VFP         0x400 /* ARM VFP/NEON registers */
-#define NT_ARM_TLS         0x401 /* ARM TLS register */
-#define NT_ARM_HW_BREAK    0x402 /* ARM hardware breakpoint registers */
-#define NT_ARM_HW_WATCH    0x403 /* ARM hardware watchpoint registers */
-#define NT_ARM_SYSTEM_CALL 0x404 /* ARM system call number */
-
-#ifndef PTRACE_GETHBPREGS
-#define PTRACE_GETHBPREGS 29
-#define PTRACE_SETHBPREGS 30
-#endif
-
-#if PTRACE_GETREGSET
-// type = 2 = write
-// static volatile uint8_t var[96] __attribute__((__aligned__(32)));
-
-static bool ll_arm64_hwbp_set(pid_t pid, ut64 _addr, int size, int wp, ut32 type) {
-	const volatile uint8_t *addr = (void *)(size_t)_addr; //&var[32 + wp];
-	const unsigned int offset = (uintptr_t)addr % 8;
-	const ut32 byte_mask = ((1 << size) - 1) << offset;
-	const ut32 enable = 1;
-	const ut32 control = byte_mask << 5 | type << 3 | enable;
-
-	struct user_hwdebug_state dreg_state = { 0 };
-	struct iovec iov = { 0 };
-	iov.iov_base = &dreg_state;
-	iov.iov_len = sizeof(dreg_state);
-
-	if (ptrace(PTRACE_GETREGSET, pid, NT_ARM_HW_WATCH, &iov) == -1) {
-		// error reading regs
-	}
-	memcpy(&dreg_state, iov.iov_base, sizeof(dreg_state));
-	// wp is not honored here i think... we can't have more than one wp for now..
-	dreg_state.dbg_regs[0].addr = (uintptr_t)(addr - offset);
-	dreg_state.dbg_regs[0].ctrl = control;
-	iov.iov_base = &dreg_state;
-	iov.iov_len = rz_offsetof(struct user_hwdebug_state, dbg_regs) +
-		sizeof(dreg_state.dbg_regs[0]);
-	if (ptrace(PTRACE_SETREGSET, pid, NT_ARM_HW_WATCH, &iov) == 0) {
-		return true;
-	}
-
-	if (errno == EIO) {
-		eprintf("ptrace(PTRACE_SETREGSET, NT_ARM_HW_WATCH) not supported on this hardware: %s\n",
-			strerror(errno));
-	}
-
-	eprintf("ptrace(PTRACE_SETREGSET, NT_ARM_HW_WATCH) failed: %s\n", strerror(errno));
-	return false;
-}
-
-static bool ll_arm64_hwbp_del(pid_t pid, ut64 _addr, int size, int wp, ut32 type) {
-	// const volatile uint8_t *addr = &var[32 + wp];
-	// TODO: support multiple watchpoints and find
-	struct user_hwdebug_state dreg_state = { 0 };
-	struct iovec iov = { 0 };
-	iov.iov_base = &dreg_state;
-	// only delete 1 bp for now
-	iov.iov_len = rz_offsetof(struct user_hwdebug_state, dbg_regs) +
-		sizeof(dreg_state.dbg_regs[0]);
-	if (ptrace(PTRACE_SETREGSET, pid, NT_ARM_HW_WATCH, &iov) == 0) {
-		return true;
-	}
-	if (errno == EIO) {
-		eprintf("ptrace(PTRACE_SETREGSET, NT_ARM_HW_WATCH) not supported on this hardware: %s\n",
-			strerror(errno));
-	}
-
-	eprintf("ptrace(PTRACE_SETREGSET, NT_ARM_HW_WATCH) failed: %s\n", strerror(errno));
-	return false;
-}
-
-static bool arm64_hwbp_add(RzDebug *dbg, RzBreakpoint *bp, RzBreakpointItem *b) {
-	return ll_arm64_hwbp_set(dbg->pid, b->addr, b->size, 0, 1 | 2 | 4);
-}
-
-static bool arm64_hwbp_del(RzDebug *dbg, RzBreakpoint *bp, RzBreakpointItem *b) {
-	return ll_arm64_hwbp_del(dbg->pid, b->addr, b->size, 0, 1 | 2 | 4);
-}
-
-#endif
-
 static int rz_debug_native_bp(RzBreakpoint *bp, RzBreakpointItem *b, bool set) {
-#if PTRACE_GETREGSET
-	if (b && b->hw) {
-		return set
-			? arm64_hwbp_add((RzDebug *)bp->user, bp, b)
-			: arm64_hwbp_del((RzDebug *)bp->user, bp, b);
-	}
-#endif
 	return false;
 }
 
@@ -782,6 +690,5 @@ static int rz_debug_desc_native_open(const char *path) {
 }
 
 static bool rz_debug_gcore(RzDebug *dbg, char *path, RzBuffer *dest) {
-	(void)path;
-	return linux_generate_corefile(dbg, dest);
+	return false;
 }

